input.go

```diff
@@ -0,0 +1,29 @@
+package shaping
+
+import "image/color"
+
+type Input interface {
+	// Text returns the characters to be shaped.
+	Text() []rune
+	// Color returns the color to render the text in.
+	Color() color.Color
+
+	// TODO should these be wrapped in an interface, eg. Font() image/font?
+
+	// FontFamily returns the name of the font family, eg. Helvetica.
+	FontFamily() string
+	// FontFace returns the name of the font face, eg. Bold Italic.
+	// TODO would it be better to split these out into Bold() bool, Italic() bool, Monospace() bool, etc?
```

Comment:
Should `Bold() bool` instead be `Weight() int`?

What about other attributes such as Oblique?

Reply:
Would things like Underline and Strikethrough be added by the shaper library or toolkit?